### PR TITLE
JwtToken auth added

### PR DESCRIPTION
**Pull request by danylivOstap**

### CODE REVIEW

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -38,7 +43,14 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                     .authenticated()
             )
             .httpBasic(Customizer.withDefaults())
+            .addFilterBefore(jwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
             .userDetailsService(userDetailsService)
             .build();
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java:53]: For JWT security in REST apps it's better to make session management stateless.

File: src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -34,13 +35,13 @@ public ErrorResponseDto handleValidationException(
     @ExceptionHandler(RegistrationException.class)
     public ErrorResponseDto handleRegistrationException(RegistrationException e) {
         return generateExceptionDetails("Registration error occurred", BAD_REQUEST,
-            List.of(e.getMessage()));
+            List.of());
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java:37]: For so called "expected" exception you can safely return message with details. Don't return exception message for exceptions that you don't expect (like handle all exception)

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/security/JwtUtil.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    public boolean isValid(String token) {
+        try {
+            Jws<Claims> claimsJws = Jwts.parserBuilder()
+                    .setSigningKey(secret)
+                    .build()
+                    .parseClaimsJws(token);
+            return !claimsJws.getBody().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            throw new JwtException("Expired or invalid jwt token");
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/security/JwtUtil.java]:
```suggestion
            return false;
```

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -8,3 +8,6 @@ spring.datasource.password=password
 spring.jpa.hibernate.show_sql=true
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.open-in-view=false
+
+jwt.expiration=3000000
+jwt.secret=superStrongSecretPassword%%%%%%%%%%%%
```
Comment (vovasalyha) [inline src/main/resources/application.properties:13]: It's better to store secrets of main application.properties in some environment/system. It's a bad prictice to push them to git. (for test application.properties you can leave any mock secrets in order for CI to pass)

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: src/main/resources/application.properties (hunk 5 above)
Comment (danylivOstap) [reply src/main/resources/application.properties:13]: Yeah, I know. I would store it in application-prod.properties. But if I creat application-dev.prorerties now, and not push them to git, no one can build this app without it.

File: src/main/java/com/example/bookstore/security/JwtUtil.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Date;
+import java.util.function.Function;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+@Getter
+@Setter
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstore/security/JwtUtil.java:19]: Do you really need these annotations?

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! 👍

**Pull request merged**